### PR TITLE
Print tips: robot_state_publisher started or not

### PR DESCRIPTION
**Pull request by dzywater**
Checkout robot_state_publisher node exists or not. If not, info: joint state publisher is not available.

### CODE REVIEW

File: webots_ros2_driver/src/WebotsNode.cpp [hunk 1]
```diff
@@ -112,8 +112,18 @@ namespace webots_ros2_driver
...
+    std::vector<std::string> nodeNames = this->get_node_names();
+    for (int k = 0; k < nodeNames.size(); ++k)
+    {
+      if (nodeNames[k] == "/robot_state_publisher")
+      {
+        setAnotherNodeParameter("robot_state_publisher", "robot_description", mRobot->getUrdf());
+        break;
+      }
+      if (k == nodeNames.size() - 1) 
+        RCLCPP_INFO(get_logger(), "No robot_state_publisher node and joint state publisher is not available.");
```
Comment (lukicdarkoo) [inline webots_ros2_driver/src/WebotsNode.cpp:124]:
```suggestion
        RCLCPP_INFO(get_logger(), "There is no the `robot_state_publisher` node, the `robot_description` parameter is not set.");
```

File: webots_ros2_driver/src/WebotsNode.cpp [hunk 2]
```diff
@@ -112,8 +112,18 @@ namespace webots_ros2_driver
 
   void WebotsNode::init()
   {
-    setAnotherNodeParameter("robot_state_publisher", "robot_description", mRobot->getUrdf());
-
+    std::vector<std::string> nodeNames = this->get_node_names();
```
Comment (lukicdarkoo) [inline webots_ros2_driver/src/WebotsNode.cpp:115]:
```suggestion
    const std::vector<std::string> nodeNames = this->get_node_names();
```

**Review by lukicdarkoo — CHANGES_REQUESTED** [review]
Nice, thanks.

I have a few style comments. Also, the `get_node_names` method returns currently available nodes. I am afraid of a situation in which the `robot_state_publisher` starts just after Webots. In general, it should not happen as Webots typically takes more time to load, but in the case of lightweight simulations, the problem may occur.

**Comment by dzywater** [comment]
> Nice, thanks.
> 
> I have a few style comments. Also, the `get_node_names` method returns currently available nodes. I am afraid of a situation in which the `robot_state_publisher` starts just after Webots. In general, it should not happen as Webots typically takes more time to load, but in the case of lightweight simulations, the problem may occur.

The second step is to provide a service which calls setAnotherNodeParameter again. Because someone may start robot_state_publisher after WebotsNode.

**Comment by lukicdarkoo** [comment]
> The second step is to provide a service which calls setAnotherNodeParameter again

That's a good idea!

File: webots_ros2_driver/src/WebotsNode.cpp (hunk 2 above)
Comment (lukicdarkoo) [reply webots_ros2_driver/src/WebotsNode.cpp:115]: Can this be a const?

**Review by lukicdarkoo — APPROVED** [review]
Thank you

**Pull request merged**